isyfact-standards-doc/src/docs/antora/modules/referenzarchitektur/pages/software-technisch/services.adoc

```diff
@@ -0,0 +1,247 @@
+
+|Korrelations-ID +
+//(deprecated)
+|X-Correlation-Id
+|ID zur Nachverfolgung von Aufrufen innerhalb einer Anwendungslandschaft.
+//Alte Benennung. +
+//*Wichtig:* Anwendungen müssen in der Übergangszeit beide Namen auflösen können.
+
+|Bearer Token
+|Authorization
+|Vorgabe des Standards OAuth 2.0.
+|===
+
+*Nutzdaten* auf der anderen Seite beinhalten alle Daten, die zur Verarbeitung eines konkreten Service-Aufrufs benötigt werden.
+Sie bilden die eigentliche, fachliche Schnittstelle und beschreiben sowohl die Daten der Anfrage sowie der Antwort.
+
+Die IsyFact standardisiert die Art und Weise, wie Nutzdaten spezifiziert, dokumentiert und technisch verarbeitet werden.
+
+[[kommunikationsarten]]
+== Kommunikationsarten
+
+IT-Systemen stehen mit der IsyFact drei Kommunikationsarten zur Verfügung: synchrone Service-Aufrufe, asynchrone Service-Aufrufe und Service-Aufrufe über Queues.
+
+
+[[synchrone-service-aufrufe]]
+=== Synchrone Service-Aufrufe
+
+Synchrone Service-Aufrufe bieten die Möglichkeit der direkten Kommunikation zwischen zwei IT-Systemen.
+Hierbei schickt der Sender eine Anfrage (englisch: _request_) an den Empfänger.
+Der Empfänger bearbeitet die Anfrage und schickt eine Antwort (englisch: _response_) an den Sender zurück.
+Der Sender wartet auf die Antwort, bevor er seine Verarbeitung fortsetzt.
+
+[[service-aufruf-synchron]]
+.Ablauf eines synchronen Service-Aufrufs
+image::software-technisch/synchroner-service-aufruf.dn.svg[]
+
+Deswegen sind synchrone Service-Aufrufe in der Regel eine vergleichsweise zeitintensive Operation.
+Häufig ist es sinnvoll, Service-Aufrufe nach Möglichkeit einzusparen.
+Das Sparen von Aufrufen kann jedoch auch Nachteile in Bezug auf Wartbarkeit bedeuten, wenn beispielsweise Redundanzen oder komplexe Caches implementiert werden müssen.
+Die Abwägung darüber muss während der Erstellung des Systementwurfs geschehen.
+
+.icon:university[title=Architekturregel] Verwendung von HTTP für Service-Aufrufe
+****
+Synchrone Service-Aufrufe finden über das Protokoll *HTTP* statt und werden sowohl zur internen als auch externen Servicekommunikation genutzt.
+HTTP-Anfragen bzw. HTTP-Antworten erlauben es an drei Stellen, anwendungsspezifische Daten zu übertragen: in der URL, in den Headern sowie im Body.
+
+[[http-message-aufbau]]
+.Aufbau von HTTP-Anfragen bzw. HTTP-Antworten
+image::referenzarchitektur:software-technisch/http-messages-aufbau.dn.svg[]
+Header enthalten Metadaten.
+Der Body enthält Nutzdaten.
+Bei Anfragen mittels `GET` und `DELETE`, die keinen Body erwarten, enthalten URL-Parameter Nutzdaten.
+****
+
+Allerdings gilt zu beachten, dass URLs (und damit auch die URL-Parameter) an vielen Stellen aufgezeichnet und in Logs geschrieben oder in Caches gehalten werden.
+Hierbei sind z.B. datenschutzrechtliche Aspekte zu prüfen, wenn URL-Parameter personenbezogene Daten enthalten.
+Im Zweifelsfall ist die Methode `POST` eine gangbare Alternative, um solche Nutzdaten im Body zu übertragen.
```

Comment:
--> In diesen Fällen wäre die Methode POST die empfohlene (oder sogar: verpflichtend zu verwendende) Alternative